pkgs/applications/office/gnucash/default.nix

```diff
@@ -1,106 +1,210 @@
+
+    cmake
+
+    makeWrapper
+
+    aqbanking
+    boost
+    glib
+    gtest
+    guile
+    gwenhywfar
+    icu
+    libdbi
+    libdbiDrivers
+    libofx
+    libxml2
+    libxslt
+    perl
+    pkg-config
+    swig
+    webkitgtk
+  ];
+
+  # GnuCash cli utils' Perl requirements
+  propagatedBuildInputs = with perlPackages; [
+    DateManip
+    FinanceQuote
+  ];
+
+  patches = [ ./0001-changes.patch ]; # this patch disables test-gnc-timezone and test-gnc-datetime which fail due to nix datetime challenges
+
+  preConfigure = ''
+    export GUILE_AUTO_COMPILE=0 # this needs to be an env variable and not a cmake flag to suppress guile warning
   '';
 
-  makeFlags = [ "GUILE_AUTO_COMPILE=0" ];
-
-  postInstall = ''
-    # Auto-updaters don't make sense in Nix.
-    rm $out/bin/gnc-fq-update
-
-    # Unnecessary in the release build.
-    rm $out/bin/gnucash-valgrind
-
-    wrapProgram "$out/bin/gnucash" \
-      --prefix XDG_DATA_DIRS : "$GSETTINGS_SCHEMAS_PATH:$out/share/gsettings-schemas/${pname}-${version}" \
-      --prefix XDG_DATA_DIRS : "${hicolor-icon-theme}/share" \
-      --prefix PERL5LIB ":" "$PERL5LIB" \
-      --set GNC_DBD_DIR ${libdbiDrivers}/lib/dbd \
-      --prefix GIO_EXTRA_MODULES : "${lib.getLib dconf}/lib/gio/modules"
-  '';
+  doCheck  = true;
 
   /*
   GNUcash's `make check` target does not define its prerequisites but expects them to have already been built.
   The list of targets below was built through trial and error based on failing tests.
   */
   preCheck = ''
-    export LD_LIBRARY_PATH=$PWD/lib:$PWD/lib/gnucash:$PWD/lib/gnucash/test:$PWD/lib/gnucash/test/future''${LD_LIBRARY_PATH:+:}$LD_LIBRARY_PATH
-    export NIX_CFLAGS_LINK="-lgtest -lgtest_main"
-    make test-scm-query test-split-register-copy-ops test-link-ofx test-import-backend test-import-account-matcher test-import-pending-matches test-qofquerycore test-import-map test-gnc-numeric test-gnc-rational test-gnc-int128 test-qofsession test-kvp-value test-gnc-guid test-numeric test-vendor test-job test-employee test-customer test-address test-business test-recurrence test-transaction-voiding test-transaction-reversal test-split-vs-account test-tokenizer test-aqb test-import-parse test-link-module-tax-us test-dynload test-agedver test-incompatdep test-modsysver test-load-c test-gnc-path-util test-xml2-is-file test-load-example-account test-query test-querynew test-lots test-group-vs-book test-account-object test-engine test-qof test-commodities test-object test-guid test-load-engine test-userdata-dir-invalid-home test-userdata-dir test-resolve-file-path test-gnc-glib-utils test-sqlbe test-column-types test-backend-dbi test-xml-transaction test-xml-pricedb test-xml-commodity test-xml-account test-string-converters test-load-backend test-kvp-frames test-dom-converters1 test-autoclear test-sx test-print-parse-amount gncmod-futuremodsys
+    make \
+      test-account-object \
+      test-address \
+      test-agedver \
+      test-app-utils \
```

Comment:
This list is totally unmaintable. Can we just run all tests and disable the few that do not work?

Reply:
I added those in https://github.com/NixOS/nixpkgs/pull/150356 when re-enabling checks. Roughly a third of the tests fail without those. The core of the problem is that it seems GnuCash expects those to just exist before running make. See https://github.com/NixOS/nixpkgs/pull/150356#issuecomment-991820624

If I understand correctly, I would have to patch those out similar to what I did in https://github.com/NixOS/nixpkgs/blob/37b0c5d39a87a231886effbde814257182b5be0c/pkgs/applications/office/gnucash/0001-changes.patch

I chose to prefer checks over no checks at the expense of the ugly list above.

Reply:
Thats everything not nice. Any other ideas?

Reply:
I could try to patch all of the test prereqs into https://github.com/Gnucash/gnucash/blob/4.9/CMakeLists.txt#L627-L630, but that's just moving the problem around within nixpkgs's domain.

Assuming that works, I can take a shot a PR'ing that into GnuCash. They take PRs but not issues on GitHub, and it looks like they want interaction via mailing lists and their Bugzilla.

I would prefer not to hold up this PR into nixpkgs for that to play out.

@SuperSandro2000 ... thoughts on what you'd like me to do?

Reply:
hmmm, well then we can leave it like it is right now even if it is ugly.

Reply:
Given all the options, leaving as-is (manually making the test prereqs) seems least bad.